app/views/contexts/summary/list_collector_block.py

```diff
@@ -0,0 +1,235 @@
+from collections import defaultdict
+from typing import Any, Mapping, Optional
+
+from flask import url_for
+from werkzeug.datastructures import ImmutableDict
+
+from app.data_models import AnswerStore, ProgressStore
+from app.data_models.list_store import ListModel, ListStore
+from app.data_models.metadata_proxy import MetadataProxy
+from app.questionnaire import Location, QuestionnaireSchema
+from app.questionnaire.placeholder_renderer import PlaceholderRenderer
+from app.questionnaire.routing_path import RoutingPath
+from app.views import contexts
+from app.views.contexts.summary.block import Block
+
+
+class ListCollectorBlock:
+    def __init__(
+        self,
+        routing_path: RoutingPath,
+        answer_store: AnswerStore,
+        list_store: ListStore,
+        progress_store: ProgressStore,
+        metadata: MetadataProxy,
+        response_metadata: Mapping,
+        schema: QuestionnaireSchema,
+        location: Location,
+        language: str,
+    ):
+        self._location = location
+        self._placeholder_renderer = PlaceholderRenderer(
+            language=language,
+            answer_store=answer_store,
+            list_store=list_store,
+            metadata=metadata,
+            response_metadata=response_metadata,
+            schema=schema,
+        )
+        self._list_store = list_store
+        self._schema = schema
+        self._location = location
+        self._section: dict = self._schema.get_section(self._location.section_id) or {}
```

Comment:
Can't be empty dict. I assume that was done for typing.
```suggestion
        self._section: dict = self._schema.get_section(self._location.section_id)
```

Reply:
Updated 👍